common/parallel/TreeCrawler.go

```diff
@@ -57,8 +57,13 @@ type Directory interface{}
 type DirectoryEntry interface{}
 
 type CrawlResult struct {
-	item DirectoryEntry
-	err  error
+	item            DirectoryEntry
+	err             error
+	enqueueToTqueue bool
```

Comment:
add a comment

//
// 'item' is a directory that needs to be added to tqueue.
// It should *not* be added to objectIndexerMap.
// This special CrawlResult will be generated by the source traverser, after it completes enumeration of a directory.
// 

to be clear we can call this enqueueDirToTqueue